ruby/lib/plus_codes/open_location_code.rb

```diff
@@ -39,40 +37,68 @@ def full?(code)
     #
     # @param latitude [Numeric] a latitude in degrees
     # @param longitude [Numeric] a longitude in degrees
-    # @param code_length [Integer] the number of characters in the code, this excludes the separator
+    # @param code_length [Integer] the number of characters in the code, this
+    # excludes the separator
     # @return [String] a plus+codes
     def encode(latitude, longitude, code_length = PAIR_CODE_LENGTH)
-      raise ArgumentError,
-      "Invalid Open Location Code(Plus+Codes) length: #{code_length}" if invalid_length?(code_length)
-
-      if code_length > MAX_CODE_LENGTH
-        code_length = MAX_CODE_LENGTH
+      if invalid_length?(code_length)
+        raise ArgumentError, 'Invalid Open Location Code(Plus+Codes) length'
       end
+
+      code_length = MAX_CODE_LENGTH if code_length > MAX_CODE_LENGTH
       latitude = clip_latitude(latitude)
       longitude = normalize_longitude(longitude)
       latitude -= precision_by_length(code_length) if latitude == 90
 
-      lat = (latitude + 90).to_r
-      lng = (longitude + 180).to_r
+      final_lat_precision = 8000 * 5**(MAX_CODE_LENGTH - PAIR_CODE_LENGTH)
```

Comment:
What's the significance of 8000? Consider moving to a constant.

Reply:
Done.

Reply:
Done.